core/commands/files.go

```diff
@@ -191,24 +225,24 @@ func statNode(ds ipld.DAGService, fsn mfs.FSNode) (*Object, error) {
 		}
 
 		var ndtype string
-		switch fsn.Type() {
-		case mfs.TDir:
+		switch d.GetType() {
+		case ft.TDirectory, ft.THAMTShard:
```

Comment:
why make this change?

Reply:
`d.GetType()` can be of [these types](https://github.com/ipfs/go-ipfs/blob/master/unixfs/format.go#L14) so if `HAMTSharding` was enabled we could potentially get an `unrecognized node type` error here, ya?

Reply:
@whyrusleeping because we want (or at least I do) support `/ipfs/Qmwhatever`. `getNodeFromPath` will return an `ipld.Node` instead of a `mfs.FSNode`.

Reply:
@frrist indeed. I got that from https://github.com/ipfs/go-ipfs/blob/master/mfs/system.go#L92 where there is the same problem. Would that be a file, directory or else ?

Reply:
I think they _must_ either be a file or a directory,
> Files is an API for manipulating IPFS objects as if they were a unix filesystem. - [file.go](https://github.com/ipfs/go-ipfs/blob/master/core/commands/files/files.go#L35)

The current definition looks good to me, what do ya'll think?

Reply:
Ideally, MFS would handle `/ipfs/` as a "mounted" subdirectory but we can fix that later.